jiant/models.py

```diff
@@ -265,6 +266,9 @@ def build_model(args, vocab, pretrained_embs, tasks):
     model = MultiTaskModel(args, sent_encoder, vocab)
     build_task_modules(args, tasks, model, d_task_input, d_emb, embedder, vocab)
     model = model.cuda() if args.cuda >= 0 else model
+    if torch.cuda.device_count() > 1:
```

Comment:
worthwhile to have a flag like `use_distributed` ? Or is the expectation that users mask devices on the commandline ?

Reply:
Because we only check for cuda_device a few times in the code I don't think it's quite necessary to have a flag for this, and yes, they can use the commandline. However, if users are on GPU, they shouldn't need to change anything.  This should automatically detect if there's more than 1 GPU, and use it if so.

Reply:
This isn't usable as-is: We have an explicit flag that asks users to specify the GPU (cuda, see comment in defaults.conf), and it seems that we're now ignoring it.

There are reasonable/common use cases in which:
- People will want to use all the GPUs on a machine. (Obvious, I guess.)
- People will want to use a specific subset of GPUs on a machine, which might not be identified by the env variable. This is common on clusters managed with older batch managers that don't track GPUs—there's often some other automatic or manual system that informs you of what GPUs you can use.
- People will want to use none of the available GPUs.

I think the simplest option would be to leave several options for CUDA:
- 'auto': Default; use all available GPUs.
- 'none' or '-1': Disable CUDA.
- '0,2,3', etc.: Use GPUs 0, 2, and 3. List would be parsed the same way as in pretrain_tasks.